### DISCUSSION TITLE
YAML Metadata blocks: passing CSS variables to HTML without switching off the smart extension?

### DISCUSSION THREAD

**Discussion by PlainMartin** [jgm/pandoc#9992] [Q&A]
I am using metadata fields in Markdown files to pass CSS attributes (e.g. colors) to an HTML template.

This will work with color _names_ or hex values, but CSS [custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties) / variables such as `var(--color-red)` will be converted to `var(–color-red)` (two hyphens converted to an n-dash) and not recognized anymore by the browser.

With `--from=markdown-smart`, the variable names are left untouched.

On the other hand, it would be nice to keep the smart extension on for, well, everything else.

So: Is there a way to pass variables such as `var(--color-red)` to an HTML template without switching off Pandoc’s [smart extension]([converted](https://pandoc.org/MANUAL.html#extension-smart))?

Thank you.

**Comment by jgm** [comment]
Anything in YAML metadata gets parsed as Markdown, so yes, `--` will be turned into an en-dash.
What you can do is mark this out as raw HTML, using the `raw_attribute` extension (enabled by default for pandoc's markdown).
e.g. `` `var(--color-red)`{=html} ``
